src/mrsimulator/method/lib/base.py

```diff
@@ -46,6 +47,16 @@ def check_rotor_frequency(cls, v, *, values, **kwargs):
             "and cannot be modified."
         )
 
+    @validator("channels", pre=True, always=True)
+    def check_for_only_one_channel(cls, v, *, values, **kwargs):
```

Comment:
The function name no longer relevant with these changes. Maybe use `check_channel_count`!